int32 angle_old
int32 angle_new